### PR TITLE
feat: adding copy button to details page

### LINKED ISSUE #4241: feat: add copy button next to title in details page
### Is your enhancement related to a problem? Please describe

I don't like selecting manually the image name especially when it is very long

### Describe the solution you'd like

Adding a copy button next to the title on details page

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by axel7083**
fixes: https://github.com/containers/podman-desktop/issues/4241

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/42176370/beb1df00-5d4c-47d9-b768-8a53d22bfe36

### CODE REVIEW

File: packages/renderer/src/lib/ui/DetailsPage.svelte [hunk 1]
```diff
@@ -17,6 +20,17 @@ function handleKeydown(e: KeyboardEvent) {
     e.preventDefault();
   }
 }
+
+let copyToolTip = 'copy';
```
Comment (lstocchi) [inline packages/renderer/src/lib/ui/DetailsPage.svelte:24]: Should it mention what you're going to copy? `Copy image name`

**Review by lstocchi — COMMENTED** [review]
LGTM, missing a test?

File: packages/renderer/src/lib/ui/DetailsPage.svelte (hunk 1 above)
Comment (axel7083) [reply packages/renderer/src/lib/ui/DetailsPage.svelte:24]: The details page is shared between many elements, such as the container, images, volumes. how am i suppose to know where is it being used ? Adding a new property that the parent should give ?

Comment (lstocchi) [reply packages/renderer/src/lib/ui/DetailsPage.svelte:24]: you could use the breadcrumb info or maybe just add a generic `copy name`

**Comment by deboer-tim** [comment]
I really like this capability & convenience, but I don't love the copy button on every page, and I think there are other things that should be copyable (e.g. ids or tags) which would mean the buttons show up beside a lot of fields.

In the Mac password dialog the user or password become links when you hover over them, and there's a popup menu to copy it. It's horrible for discovery (yet I found it easily enough) but also means you can click anywhere on the field. Maybe we don't even need a popup and copy whenever you click on the hover? I don't know, just some ideas to reduce visual clutter and make copying as easy as possible.

@mairin any thoughts on the best way to enable this?

**Comment by axel7083** [comment]
> I really like this capability & convenience, but I don't love the copy button on every page, and I think there are other things that should be copyable (e.g. ids or tags) which would mean the buttons show up beside a lot of fields.
> 
> In the Mac password dialog the user or password become links when you hover over them, and there's a popup menu to copy it. It's horrible for discovery (yet I found it easily enough) but also means you can click anywhere on the field. Maybe we don't even need a popup and copy whenever you click on the hover? I don't know, just some ideas to reduce visual clutter and make copying as easy as possible.
> 
> @mairin any thoughts on the best way to enable this?

From my point of view, I am always copying either images  names or container names, and the current way is to select the text then copy past, for very long names, it always takes a few seconds and makes it pretty repetitive. 

I do not like the personally the idea of having a popup for copying elements, the hover tooltip is pretty self explanatory in this context, and the icon used here is also pretty common, therefore confirmation for clicking on a copy icon is probably not useful.

The idea of making the title a link, with a hover message copy, thus doing the same, is also a good solution. In the same way as the button, if a hover tooltip show "Copy content" or something similar, it should not show a popup making it a procedure as long as selecting and copying manually.

Just tell me which solution you would prefer, and I will adapt the code

**Comment by axel7083** [comment]
> LGTM, missing a test?

Done!

**Comment by mairin** [comment]
> > @mairin any thoughts on the best way to enable this?

Sorry, I had it on my list to do a quick mockup for this but time got away from me -

my thinking was to have the clipboard icon appear on hover / kb focus - so you wont have the visual clutter and it'll scale visually. and if its something to copy your instinct might ve to try to select + copy, so you'll naturally discover the hover action enroute to that.

what do you think

**Comment by cdrage** [comment]
> > > @mairin any thoughts on the best way to enable this?
> 
> Sorry, I had it on my list to do a quick mockup for this but time got away from me -
> 
> my thinking was to have the clipboard icon appear on hover / kb focus - so you wont have the visual clutter and it'll scale visually. and if its something to copy your instinct might ve to try to select + copy, so you'll naturally discover the hover action enroute to that.
> 
> what do you think

Sums up from me too, @axel7083 sorry for the long wait, are you able to update the PR with @mairin 's suggestion and let's get this merged in?

**Comment by axel7083** [comment]
> > > @mairin any thoughts on the best way to enable this?
> 
> Sorry, I had it on my list to do a quick mockup for this but time got away from me -
> 
> my thinking was to have the clipboard icon appear on hover / kb focus - so you wont have the visual clutter and it'll scale visually. and if its something to copy your instinct might ve to try to select + copy, so you'll naturally discover the hover action enroute to that.
> 
> what do you think

I am not quick sure to get what you mean by appear on hover:
- Do you mean on the tooltip ? (Then the current Tooltip implementation only support String so it cannot be added).
- Do you mean having literally the mouse icon to be the copy icon ? (Very complicated to do, so i'm guessing this is really not that)
- Adding the copy icon on the right of the title while it is hovering the title, with a tooltip with "Copy" text.

I went with the last option (see video in the Edit section above)

**Comment by trmendes** [comment]
Not planing on changing the flow of this conversation, but I would like to share my feelings on the topic.

I spent some time checking what is the docker desktop approach for providing the user a way to copy content, and it same we do. They have “copy to clipboard” buttons. Difference is that the copy button is not so bright as ours. They usually make it more gray.

I personally think we could just be more careful choosing the places to have such copy button and, find a more soft way (can't find a better word for it) to present such icon. 

The option to have a different approach is also nice, but I also struggle to fully grasp the concept.

What comes to my mind is a pop-up (tooltip) with a copy button on it, just like the ones we have on mobile platforms.

If that is that case, I would be confused on what is the content to be copied: The hovered content or the text that is shown in the tooltip, together with the copy button.

@deboer-tim maybe you could post a screenshot or a video? :D

**Comment by deboer-tim** [comment]
Hi @trmendes. I tried to throw a quick sample together twice and it didn't quite work, so no video. :) My fallback thought was that we should make a /ui component that puts a copy button on the right for now, with clean enough API that if we ever came up with another design we could easily change it in the future without changing all the places that would use it.

I just looked at what you did in #5687 (thank you, btw), and CopyToClipboard is basically that. 👍🏼 We need to use it in a few more places to see if there are any tweaks necessary and it works well anywhere, but that's a great start.

**Comment by trmendes** [comment]
@deboer-tim Thank you for your answer. Do we have an issue listing the places we would like to make usagge of ```CopyToClipboard``` component?

A quick search for 'Clipboard' on main:

```
TroubleshootingDevToolsConsoleLogs.svelte
EmptyScreen.svelte
```

**Comment by deboer-tim** [comment]
I'm only aware of these two issues open (#4241 and #5608), but I think I've seen a couple others in messages/forums, and IMHO any sha/url could be requested at some point. So we can start with these two and get some feedback/use and see if others are requested.

**Pull request closed without merging**